### PR TITLE
Change prog ID to int and define hooks in BPF

### PR DESCRIPTION
**Pull request by mazdakn**
## Description
This PR includes two changes:
- Use `int` for program ID.
- Define bpf hooks in BPF package and use it for all other ones.
- Move `MapPinPath` function to bpf package to use it in XDP.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf/bpf.go [hunk 1]
```diff
@@ -2268,3 +2280,24 @@ const countersMapVersion = 1
 func CountersMapName() string {
 	return fmt.Sprintf("cali_counters%d", countersMapVersion)
 }
+
+func MapPinPath(typ int, name, iface string, hook Hook) string {
+	PinBaseDir := path.Join(DefaultBPFfsPath, "tc")
+	subDir := "globals"
+	if (typ == unix.BPF_MAP_TYPE_PROG_ARRAY && strings.Contains(name, JumpMapName())) ||
```
Comment (song-jiang) [inline felix/bpf/bpf.go:2287]: Suggest to add comments why these two maps require special subdir.

File: felix/dataplane/linux/bpf_ep_mgr.go [hunk 2]
```diff
@@ -1882,7 +1883,7 @@ func FindJumpMap(progIDStr, ifaceName string) (mapFD bpf.MapFD, err error) {
 		}
 	}
 
-	return 0, fmt.Errorf("failed to find jump map for iface=%v progID=%v", ifaceName, progIDStr)
+	return 0, fmt.Errorf("failed to find jump map for iface=%v progID=%v", ifaceName, progID)
```
Comment (song-jiang) [inline felix/dataplane/linux/bpf_ep_mgr.go:1886]: replace `%v` with `%s` and `%d`?

**Review by song-jiang — APPROVED** [review]
Just a few nits but LGTM

**Review by sridhartigera — APPROVED** [review]
LGTM

**Pull request merged**